third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp

```diff
@@ -121,11 +121,19 @@ class LoadStorePrefetchOpConversion
            "only support 1d/2d load/store/prefetch for now");
 
     unsigned dataSize = tensorType.getElementType().getIntOrFloatBitWidth();
-    unsigned blockWidth = tensorType.getShape()[1];
-    assert(blockWidth == 16 || blockWidth == 32 && "only support 16/32 block");
-    unsigned vBlks = blockWidth == 32 ? 2 : 1;
-    blockWidth = 16;
     unsigned blockHeight = tensorType.getShape()[0];
+    unsigned blockWidth = tensorType.getShape()[1];
+    unsigned vBlks = 1;
+    if (dataSize == 16) {
+      vBlks = blockWidth / 16;
+      blockWidth = 16;
+    } else if (dataSize == 8 && (blockHeight == 16 || blockHeight == 32)) {
+      unsigned blockWidthUnit = blockHeight == 16 ? 32 : 16;
+      vBlks = blockWidth / blockWidthUnit;
```

Comment:
should we use ceil to avoid getting zero?
```suggestion
      vBlks = ceil(blockWidth, blockWidthUnit);
```

Reply:
Done.